pkg/cmd/extension/command.go

```diff
@@ -220,9 +228,83 @@ func checkValidExtension(rootCmd *cobra.Command, m extensions.ExtensionManager,
 	return nil
 }
 
+func isBinExtension(client *http.Client, repo ghrepo.Interface) (isBin bool, err error) {
+	var r *release
+	r, err = fetchLatestRelease(client, repo)
+	if err != nil {
+		httpErr, ok := err.(api.HTTPError)
+		if ok && httpErr.StatusCode == 404 {
+			err = nil
+			return
+		}
+		return
+	}
+
+	for _, a := range r.Assets {
+		dists := possibleDists()
+		for _, d := range dists {
+			if strings.HasSuffix(a.Name, d) {
+				isBin = true
+				break
+			}
+		}
+	}
+
+	return
+}
+
 func normalizeExtensionSelector(n string) string {
 	if idx := strings.IndexRune(n, '/'); idx >= 0 {
 		n = n[idx+1:]
 	}
 	return strings.TrimPrefix(n, "gh-")
 }
+
+func possibleDists() []string {
```

Comment:
I didn't want to force authors to support a default OS-ARCH -- an extension might only make sense on windows, for example, because it integrates with the registry editor or something. What's your concern with the list?

Reply:
I guess the list is fine. It's just that the list feels long and it's not clear which combos of OS-ARCH made this cut in the first place and whether the list should be kept up to date with additional OS-ARCH combos in the future.

Reply:
it's the full list that go supports as output by `go tool dist list`. i figure new combinations are added so rarely that it wouldn't be an issue.